src/Functions/sqid.cpp

```diff
@@ -80,21 +82,86 @@ class FunctionSqid : public IFunction
     sqidscxx::Sqids<> sqids;
 };
 
+/// sqidDecode(number1, ...)
+class FunctionSqidDecode : public IFunction
+{
+public:
+    static constexpr auto name = "sqidDecode";
+
+    String getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 1; }
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionSqidDecode>(); }
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        FunctionArgumentDescriptors args{
+            {"sqid", &isString<IDataType>, nullptr, "String"}
+        };
+        validateFunctionArgumentTypes(*this, arguments, args);
+
+        return std::make_shared<DataTypeArray>(std::make_shared<DataTypeUInt64>());
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        auto col_res_nested = ColumnUInt64::create();
+        auto & res_nested_data = col_res_nested->getData();
+
+        auto col_res_offsets = ColumnArray::ColumnOffsets::create();
+        auto & res_offsets_data = col_res_offsets->getData();
+        res_offsets_data.reserve(input_rows_count);
+
+        for (size_t i = 0; i < input_rows_count; ++i)
+        {
+            const ColumnWithTypeAndName & arg = arguments[0];
+            ColumnPtr current_column = arg.column;
+            std::string_view sqid = current_column->getDataAt(i).toView();
```

Comment:
Maybe better to use specific `ColumnString` here instead of calling virtual `getDataAt`? Or it's not a big deal?

Reply:
Sure, why not, at least the compiler can devirtualize then (`ColumnString` is marked `final`).